components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41.java

```diff
@@ -426,10 +427,21 @@ public Response getCartridgeByFilter(
     @AuthorizationAction("/permission/stratos/cartridges/manage")
     public Response removeCartridge(
             @PathParam("cartridgeType") String cartridgeType) throws RestAPIException {
-        StratosApiV41Utils.removeCartridge(cartridgeType);
-        return Response.ok().entity(new ResponseMessageBean(ResponseMessageBean.SUCCESS,
-                String.format("Cartridge deleted successfully: [cartridge-type] %s", cartridgeType))).build();
-
+        try {
+            StratosApiV41Utils.removeCartridge(cartridgeType);
+            return Response.ok().entity(new ResponseMessageBean(ResponseMessageBean.SUCCESS,
+                    String.format("Cartridge deleted successfully: [cartridge-type] %s", cartridgeType))).build();
+        } catch (RemoteException e) {
```

Comment:
Let's not send BAD_REQUEST for Remote exception. We can throw a RestAPIException for that, since it will be an internal server error.